en, zh: add scheduler scheduling based on customer topology (#483)
pingcap/docs-tidb-operator: en/tidb-scheduler.md

A TiDB cluster includes three key components: PD, TiKV, and TiDB. Each consists of multiple nodes: PD is a Raft cluster, and TiKV is a multi-Raft group cluster. PD and TiKV components are stateful. The default scheduling rules of the Kubernetes scheduler cannot meet the high availability scheduling requirements of the TiDB cluster, so the Kubernetes scheduling rules need to be extended.

> Currently, pods can be scheduled according to specific dimensions in TidbCluster annotations, such as:

Reviewer: ```suggestion
Currently, pods can be scheduled according to specific dimensions in TidbCluster `metadata.annotations`, such as:
```

> Currently, pods can be scheduled according to specific dimensions in TidbCluster `metadata.annotations`, such as:

Reviewer 2: ```suggestion
Currently, pods can be scheduled according to specific dimensions by modifying `metadata.annotations` in TidbCluster, such as:
```

{{< copyable "" >}}

```yaml
metadata:
  annotations:
    pingcap.com/ha-topology-key: kubernetes.io/hostname
```

> Or modify tidb-Cluster Chart `values.yaml`:

Reviewer 2: ```suggestion
Or by modifying tidb-cluster Chart `values.yaml`:
```

{{< copyable "" >}}

```yaml
haTopologyKey: kubernetes.io/hostname
```

> The above configuration indicates scheduling by node (default value) dimensions. If you want to schedule pods by other dimensions, 
> such as: 'pingcap.com/ha-topology-key: zone', which schedules by zone,
> Each node should also be labeled as follows:

Reviewer 2: ```suggestion
The configuration above indicates scheduling by the node dimension (default). If you want to schedule pods by other dimensions, such as `pingcap.com/ha-topology-key: zone`, which means scheduling by zone, each node should also be labeled as follows:
```

{{< copyable "shell-regular" >}}

```shell
kubectl label nodes node1 zone=zone1
```

> Different nodes may have different labels or the same label, and if a node is not labeled, the scheduler will not schedule pod to that node.

Reviewer 2: ```suggestion
Different nodes may have different labels or the same label, and if a node is not labeled, the scheduler will not schedule any pod to that node.
```

> TiDB Scheduler implements the following customized scheduling rules(The following example is based on node scheduling, scheduling rules based on other dimensions are the same):

Reviewer 2: ```suggestion
TiDB Scheduler implements the following customized scheduling rules. The following example is based on node scheduling, scheduling rules based on other dimensions are the same.
```
